nlp_primitives/lsa.py

```diff
@@ -80,3 +99,34 @@ def lsa(array):
             return pd.Series(arr)
 
         return lsa
+
+
+def _validate_trainer(trainer):
```

Comment:
If we're expecting a specific pipeline structure here, instead of exposing a `trainer`parameter as input, I think we should expose a `training_corpus` parameter instead:
```python
lsa = LSA(training_corpus=corpus)
```

Then we can replace this validation with validation of the input data based on the expectations for that.

Which brings up a different question: what are the requirements on the input to TfidfVectorizer? I'm assuming it expects a single column containing text, can handle either pd dataframe or np array, etc.

Finally, I think we should define the validation/make methods inside the class, to show they're specific to this class

Reply:
I like the idea of passing in the corpus instead of the trainer object here. I didn't go with that option for this implementation due to featuretools' requirement that any values passed into the `__init__` of a primitive must be saved as a class attribute, which leads to unnecessary storage of the entire corpus.

I'm starting to lean towards moving this implementation to evalml instead. I don't like having these functions floating around outside the class, as there is no sort of structure in place for what they should look like yet. @rwedge would the same `__init__` --> class attribute requirement hold if this primitive was implemented outside of featuretools, or would there be any way to work around it? If so, that would be another bonus to moving this all to evalml instead.

Reply:
@eccabay ah, I didn't know about that requirement. Makes sense. In that case, I agree, let's move this into an LSA component in evalml, instead of making the changes here and using the primitive.

Reply:
Oops, I meant an instance attribute

Reply:
The new plan sounds good, I'll close this PR and move the implementation over to evalml.